addons/sourcemod/scripting/vsh/bosses/boss_uberranger.sp

```diff
@@ -402,8 +432,96 @@ public void UberRangerResetColorList()
 	g_aUberRangerColorList.PushArray({ 79, 100, 59 }); 		// Zepheniah's Greed (modified)
 
 	g_aUberRangerColorList.PushArray({ 25, 230, 230 }); 	// Cyan
+	g_aUberRangerColorList.PushArray({ 100, 100, 100 }); 	// Gray
 	g_aUberRangerColorList.PushArray({ 255, 110, 0 }); 		// Orangered
 	g_aUberRangerColorList.PushArray({ 88, 21, 132 }); 		// Purple (dark)
 	
 	g_aUberRangerColorList.Sort(Sort_Random, Sort_Integer);
+}
+
+public int UberRanger_SpawnBestPlayer(ArrayList aClients)
+{
+	int iBestClientIndex = -1;
+	int iLength = aClients.Length;
+	int iBestScore = -1;
+	
+	for (int i = 0; i < iLength; i++)
+	{
+		int iClient = aClients.Get(i);
+		
+		if (!g_bUberRangerPlayerWasSummoned[iClient])
+		{
+			int iClientScore = SaxtonHale_GetScore(iClient);
+			
+			if (iClientScore > iBestScore)
+			{
+				iBestScore = iClientScore;
+				iBestClientIndex = iClient;
+			}
+		}
+	}
+
+	if (iBestClientIndex > 0)
+	{
+		SaxtonHaleBase boss = SaxtonHaleBase(iBestClientIndex);
+		if (boss.bValid)
+			boss.CallFunction("Destroy");
+
+		//Allow them to join the boss team
+		Client_AddFlag(iBestClientIndex, ClientFlags_BossTeam);
+		TF2_ForceTeamJoin(iBestClientIndex, TFTeam_Boss);
+
+		boss.CallFunction("CreateBoss", "CMinionRanger");
+		TF2_RespawnPlayer(iBestClientIndex);
+
+		//Duration of this condition will reset when they move
+		TF2_AddCondition(iBestClientIndex, TFCond_UberchargedCanteen, 7.0);
+	}
+	
+	//Returns -1 if it finds nobody suitable
+	return iBestClientIndex;
+}
+
+public Action Timer_UberRanger_ReplaceMinion(Handle hTimer, int iClient)
+{
+	if (hTimer != g_hUberRangerMinionAFKTimer[iClient])
+		return;
+		
+	if (TF2_GetClientTeam(iClient) < TFTeam_Attack || !IsPlayerAlive(iClient) || g_bUberRangerMinionHasMoved[iClient])
+		return;
+	
+	//Adjust the countdown on screen
+	if (g_iUberRangerMinionAFKTimeLeft[iClient] > 0)
+	{
+		g_iUberRangerMinionAFKTimeLeft[iClient]--;
+		g_hUberRangerMinionAFKTimer[iClient] = CreateTimer(1.0, Timer_UberRanger_ReplaceMinion, iClient);
+		return;
+	}
+	
+	//Snap the AFK player. There's no point in killing them if they're the only acceptable client available
+	ArrayList aValidMinions = new ArrayList();
+	GetValidSummonableClients(aValidMinions);
```

Comment:
Why not create new `ArrayList` inside `GetValidSummonableClients` without needing to pass handle to param